fakenet/diverters/diverterbase.py

```diff
@@ -1782,3 +1806,207 @@ def maybeExecuteCmd(self, pkt, pid, comm):
+            N::::::N       N:::::::NB:::::::::::::::::B I::::::::Is::::::::::::::s 
+            N::::::N        N::::::NB::::::::::::::::B  I::::::::I s:::::::::::ss  
+            NNNNNNNN         NNNNNNNBBBBBBBBBBBBBBBBB   IIIIIIIIII  sssssssssss    
+                                                                       
+                                                                                                               
+            ========================================================================
+                                Network-Based Indicators Summary
+            ========================================================================
+        """
+        indent = "  "
+        self.logger.info(banner)
+        process_counter = 0
+        for process_info, values in self.nbis.items():
+            process_counter += 1
+            self.logger.info(f"[{process_counter}] Process ID: "
+                    f"{process_info[0]}, Process Name: {process_info[1]}")
+
+            for application_layer_proto, nbi_entry in values.items():
+                self.logger.info(f"{indent*2} Protocol: "
+                                 f"{application_layer_proto}")
+                nbi_counter = 0
+
+                for attributes in nbi_entry:
+                    nbi_counter += 1
+                    self.logger.info(f"{indent*3}{nbi_counter}.Transport Layer "
+                                     f"Protocol: {attributes['transport_layer_proto']}")
+                    self.logger.info(f"{indent*4}Source port: {attributes['sport']}")
+                    self.logger.info(f"{indent*4}Destination IP: {attributes['dst_ip']}")
+                    self.logger.info(f"{indent*4}Destination port: {attributes['dport']}")
+                    self.logger.info(f"{indent*4}SSL encrypted: "
+                                     f"{attributes['is_ssl_encrypted']}")
+                    self.logger.info(f"{indent*4}Network mode: "
+                                     f"{attributes['network_mode']}")
+
+                    for key, v in attributes['nbi'].items():
+                        if v is not None:
+                            # Let's convert the NBI value to str if it's not already
+                            if isinstance(v, bytes):
+                                v = v.decode('utf-8')
+
+                            # Let's print maximum 40 characters for NBI values
+                            v = (v[:40]+"...") if len(v)>40 else v
+                        self.logger.info(f"{indent*6}-{key}: {v}")
+
+                    self.logger.info("\r")
+
+            self.logger.info("\r")
+
+    def generate_html_report(self):
+        """Generates an interactive HTML report containing NBI summary saved
+        to the main working directory of flare-fakenet-ng. Called by stop() method
+        of diverter.
+        """
+        TEMPLATE_FILE = os.path.join("fakenet", "configs", "html_report_template.html")
```

Comment:
The thought was to keep TEMPLATE_FILE as constant so that it could be used in other methods like ones to generate the communication graph. But currently, it is used in just one method. So, we will use normal naming convention. Thanks for pointing out :)